doc/release/1.13.0-notes.rst

```diff
@@ -22,6 +22,8 @@ Dropped Support
 Deprecations
 ============
 
+* ``in1d`` is now deprecated in favor of ``isin``.
```

Comment:
I don't think we should mark things as deprecated without also raising a `DeprecationWarning`.

I think it would be best to not deprecate `in1d` in this PR, so that merging this otherwise nice work can happen in time for 1.13.

You can always make a new PR that adds deprecation once this is merged